server.js

```diff
@@ -985,6 +984,46 @@ routes['join_room'] = function (user, data) {
     room.sync(user);
 }
 
+routes['search_youtube'] = function (user, data) {
+	if(user.room === null) return;
+	var room = user.room;
+
+	var query = encodeURIComponent(data.query);
+	if (typeof(data.query === 'string') && query.length > 0 && query.length < 101) {
```

Comment:
data is a javascript object straight from the client, so it can contain anything. Perhaps in the future we could adopt protobuf to avoid these kinds of checks.